polaris-migrator/src/migrations/scss-remove-unused-import/scss-remove-unused-import.ts

```diff
@@ -0,0 +1,58 @@
+import type {FileInfo, API, Options} from 'jscodeshift';
+import postcss from 'postcss';
+import type {Plugin} from 'postcss';
+
+import {createInlineComment} from '../../utilities/sass';
+import {POLARIS_MIGRATOR_COMMENT} from '../../constants';
+
+interface PluginOptions extends Options {
+  importPath?: string;
+  namespace?: string;
+}
+
+const plugin = (options: PluginOptions = {}): Plugin => {
+  const {
+    importPath = 'global-styles/legacy-polaris-v8',
+    namespace = 'legacy-polaris-v8',
+  } = options;
+  const namespacePrefix = `${namespace}.`;
+
+  return {
+    postcssPlugin: 'scss-remove-unused-imports',
+    Once(root) {
+      root.walkAtRules('use', (atRule) => {
+        const usePath = atRule.params.match(/(['"])(.*?)\1/)?.[2] ?? '';
+
+        // Didn't find the target import in file
+        if (!usePath.endsWith(importPath)) {
```

Comment:
Thoughts on using our `matchesStringOrRegExp` util here? The `importPath` option doesn't naturally communicate `endsWith` (IMO) and would still be achievable with the following pattern `/global-styles\/my-module$/`.